src/components/VisualizationStep.tsx

```diff
@@ -96,7 +96,7 @@ const VisualizationStep = ({ data }: NodeProps<IVizStepNodeData>) => {
           data-testid={`viz-step-${data.step.name}`}
         >
           {/* PREPEND STEP BUTTON */}
-          {VisualizationService.showPrependStepButton(data, endStep) && (
+          {VisualizationService.showPrependStepButton(data, endStep, visualizationStore.nodes) && (
```

Comment:
This could be changed to an instance method `visualizationService.showPrependStepButton(data)` then the method can check if it's endStep, and retrieve the `visualizationStore.nodes` from its reference, WDYT?

Reply:
Thanks @igarashitm , some questions:
1. Do you mean to instead check the `node.data.step.type` directly from `showPrependStepButton()`, basically what the `StepsService.isEndStep()` method does, but manually? 
2. If I remove the `visualizationStore.nodes` parameter how would I access the `visualizationStore`? It always tells me it doesn't exist on `VisualizationService`.

Reply:
1. I think it's even better if `showPrependStepButton()` could just rely on `StepsService.isEndStep()`
2. ah so this component doesn't have the VisualizationService instance, I would do `const visualizationService = new VisualizationService(...)` just like this - https://github.com/KaotoIO/kaoto-ui/pull/1270/files#diff-714a52075514a965fd07e5a9d53ace053b12ff84b11f2c91f9041aa1124c3e3cL25

Reply:
Acutually like this - https://github.com/KaotoIO/kaoto-ui/blob/main/src/components/Visualization.tsx#L42

Reply:
and don't forget to remove `static` from `showPrependStepButton()`

Reply:
ahh how silly, I didn't see we had no instantiated it already from the component. Thank you, these are good ideas, I'll make the changes now.

Reply:
oops only one question--how can I instantiate the service from the unit tests? then we would have to mock the store as well, right?

Reply:
and then it complains because obviously I'm not using the `visualizationService` value, though maybe I don't need to assign it to anything and just do `new visualizationService()`, and unit tests cannot access if it's not `static`.

Reply:
This is for `StepsService` though, I could instantiate with the real store (but empty) - https://github.com/KaotoIO/kaoto-ui/blob/main/src/services/stepsService.test.ts#L9-L13